pymc_experimental/marginal_model.py

```diff
@@ -251,9 +253,24 @@ class FiniteDiscreteMarginalRV(MarginalRV):
     """Base class for Finite Discrete Marginalized RVs"""
 
 
+def static_shape_ancestors(vars):
```

Comment:
So before https://github.com/pymc-devs/pytensor/commit/0cb5fdbd6c3ec58cf0172db91041098f6e175ab7, these Shape Ops would have been caught by a the simple `isinstance(var, Constant)` check on line 276?

Now a "constant" shape is one which 1) has no `None`s, and 2) has only `Shape` `Op` ancestors?

Reply:
Before there wouldn't be any dependency between x -> shape(x) -> y(shape(x)), because `shape(x)` would just be a constant (so no further edges to x)